src/main/java/aeneas/views/BuildLevelView.java

```diff
@@ -83,16 +121,92 @@ public void initialize(URL location, ResourceBundle resources) {
         new Square(1, 4),
     });
 
-
     bullpenView.addPiece(testPiece);
 
     boardView = new BoardView(levelModel.getBoard());
     VBox.setMargin(boardView, new Insets(10, 10, 10, 10));
     centerBox.setAlignment(Pos.TOP_RIGHT);
     centerBox.getChildren().add(boardView);
 
-    saveButton.setOnMouseClicked(
-        new SaveLevelController(parentView, levelModel));
+    saveButton.setOnMouseClicked(new SaveLevelController(parentView));
+
+    // Handle changes in the level type.
+    // TODO: Consider moving this to a separate class.
+    levelType.selectedToggleProperty().addListener((ObservableValue<? extends Toggle> ov,
+                Toggle toggle, Toggle new_toggle) -> {
+      String levelName = ((RadioButton)new_toggle).getText();
+      LevelType type = LevelType.fromString(levelName);
```

Comment:
So, instead of defining the strings in the FXML and doing reverse lookup we should populate the list of level types from code.